src/braket/analog_hamiltonian_simulator/rydberg/rydberg_simulator.py

```diff
@@ -0,0 +1,198 @@
+            rydberg_interaction_coef
+        ) / ((SPACE_UNIT**6) / TIME_UNIT)
+        self.blockade_radius = validate_blockade_radius(blockade_radius) / SPACE_UNIT
+
+        # Convert the units of the program from SI unit to microsecond and micrometer etc.
+        program = convert_unit(program)
+
+        # Get the duration of the program. Note that there could be 0 or 1 driving field
+        # (same for the shifting field) for a given program. Hence we need to checkout
+        # the duration for both fields
+        self.duration = 0
+        if len(program.hamiltonian.drivingFields) == 1:
+            self.duration = float(
+                program.hamiltonian.drivingFields[0].amplitude.time_series.times[-1]
+            )
+        elif len(program.hamiltonian.shiftingFields) == 1:
+            self.duration = float(
+                program.hamiltonian.shiftingFields[0].magnitude.time_series.times[-1]
+            )
+
+        if self.duration == 0:
+            self.simulation_times = [0]
+        else:
+            self.simulation_times = np.linspace(0, self.duration, steps)
+
+        # Get valid configurations that comply with the blockade approximation
+        self.atomArray = program.setup.ahs_register
+        self.configurations = get_blockade_configurations(self.atomArray, self.blockade_radius)
+
+        # Run the solver
+        # We shall adaptive change between numpy solver (RK6 method) and scipy solver
+        if len(self.configurations) <= 1000:
+            states = rk_run(
+                program,
+                self.configurations,
+                self.simulation_times,
+                self.rydberg_interaction_coef,
+            )
+        else:
+            states = scipy_integrate_ode_run(
+                program,
+                self.configurations,
+                self.simulation_times,
+                self.rydberg_interaction_coef,
+                progress_bar=progress_bar,
+                atol=atol,
+                rtol=rtol,
+                solver_method=solver_method,
+                order=order,
+                nsteps=nsteps,
+                first_step=first_step,
+                max_step=max_step,
+                min_step=min_step,
+            )
+
+        # Convert the result type
+        if shots == 0:
+            raise NotImplementedError("Shot = 0 is not currently implemented")
+        else:
+            dist = sample_state(states[-1], shots)
+            return convert_result(
+                dist,
+                self.atomArray.filling,
+                self.configurations,
+                self.mock_taskMetadata(shots),
+            )
+
+    def mock_taskMetadata(self, shots):
+        return TaskMetadata(
+            id="rydberg",
+            shots=shots,
+            deviceId="rydbergLocalSimulator",
+        )
+
+    @property
+    def properties(self) -> DeviceCapabilities:
+        """simulator properties"""
+        mock_dict = {
```

Comment:
mock_dict seems like a misleading name here, we're not mocking anything

Reply:
This property is a temporary solution where I will need the `RydbergSimulatorDeviceCapabilities.action` to make the local simulator run at the SDK level. In this sense, it is mocking the dictionary here https://github.com/aws/amazon-braket-default-simulator-python/blob/c542653d10704adbc1d237284a09e89f065dc014/src/braket/default_simulator/state_vector_simulator.py#L54, for example. I can change the name here if needed, but I think we should make the neutral atom simulator look closer to other simulators at the SDK. Let me know what you think.

Reply:
No I am not saying that, and in fact I don't know how this particular property will/should look like. Let me explain. 
1. If we look at this line in the SDK, we will need `simulator.properties.action` just like other simulators, that is why I define this property with this `mock_dict` that contains `"action": {"braket.ir.ahs.program": {}}`. 
2. If we look at other simulators, there is dedicated capability classes in here https://github.com/aws/amazon-braket-schemas-python/tree/quera/src/braket/device_schema/simulators, which does not have the capability class for AHS simulator yet. 
3. That is why I introduced `RydbergSimulatorDeviceCapabilities` by hand here. But this should be just a temporary solution, right? I don't know what is the plan for this part. 
4. The local simulator should be independent from the real device, hence it should not need the real device capability. 

Let me know if this makes sense.

Reply:
I think this approach is fine for now, but let's rename the variable. We can call it properties, or even just put the definition inline and bypass naming it altogether. Then when we have action properties, we can populate the empty dict in the action field here